src/main/java/edu/umsystem/SortShow.java

```diff
@@ -109,22 +80,9 @@ public void paintComponent(Graphics g) {
         super.paintComponent(g);
         //A loop to assign a colour to each line
         for (int i = 0; i < total_number_of_lines; i++) {
-            //using eight colours for the lines
-            if (i % 8 == 0) {
-                g.setColor(Color.green);
-            } else if (i % 8 == 1) {
-                g.setColor(Color.blue);
-            } else if (i % 8 == 2) {
-                g.setColor(Color.yellow);
-            } else if (i % 8 == 3) {
-                g.setColor(Color.red);
-            } else if (i % 8 == 4) {
-                g.setColor(Color.black);
-            } else if (i % 8 == 5) {
-                g.setColor(Color.orange);
-            } else if (i % 8 == 6) {
-                g.setColor(Color.magenta);
-            } else g.setColor(Color.gray);
+
+            //Set color hue according to height
+            g.setColor(Color.getHSBColor((float) lines_lengths[i] / total_number_of_lines,1,1));
 
             //Drawing the lines using the x and y-components
             g.drawLine(4 * i + 25, 300, 4 * i + 25, 300 - lines_lengths[i]);

```

Comment:
ooh, I love the Hue idea!! I personally was having some difficulty seeing it with 1px wide lines though so what if we also switched the drawLine out for something like this?

```suggestion
            g.fillRect(4 * i + 25, 300 - lines_lengths[i], 4, lines_lengths[i]);
```